packages/integration-karma/test/synthetic-shadow/inner-outer-text/inner-outer-text.spec.js

```diff
@@ -0,0 +1,147 @@
+import { createElement } from 'lwc';
+import Container from 'x/container';
+
+if (!process.env.NATIVE_SHADOW) {
+    describe('innerText', () => {
+        const elm = createElement('x-container', { is: Container });
+        document.body.appendChild(elm);
+
+        it('should remove consecutive LF in between from partial results', () => {
+            const testCase = elm.shadowRoot.querySelector('.consecutive-LF');
+
+            expect(testCase.innerText).toBe(`initial
+
+first case text
+
+second case text
+
+end`);
+        });
+
+        it('should remove hidden text + removes empty text between LF counts', () => {
+            const testCase = elm.shadowRoot.querySelector('.hidden-text-in-between');
+
+            expect(testCase.innerText).toBe(`initialend`);
+        });
+
+        it('should remove hidden text with css', () => {
+            const testCase = elm.shadowRoot.querySelector('.hidden-text-with-css');
+
+            expect(testCase.innerText).toBe(`initialend`);
+        });
+
+        it('should display textContent if element is hidden', () => {
+            const testCase = elm.shadowRoot.querySelector('.element-hidden-shows-text-content');
+
+            expect(testCase.innerText).toBe(`initialfirst case textsecond case textend`);
+        });
+
+        it('should return empty when childNodes are hidden', () => {
+            const testCase = elm.shadowRoot.querySelector('.empty-inner-text-due-hidden-children');
+
+            expect(testCase.innerText).toBe(``);
+        });
+
+        it('should collect text from multiple levels', () => {
+            const testCase = elm.shadowRoot.querySelector('.collect-text-multiple-levels');
+
+            expect(testCase.innerText)
+                .toBe(`This is, a text that should be displayed, in one line. It includes links.
+
+Also paragraphs
+
+and then another text`);
+        });
+
+        it('should collect text from tables (table-cell and table-row)', () => {
+            const testCase = elm.shadowRoot.querySelector('.table-testcase');
+
+            // Notice that:
+            // 1. the last \t on each row is incorrect, it's a relaxation from the spec.
+            // 2. the last \n is incorrect, it's also a relaxation from the spec.
+            expect(testCase.innerText).toBe(`1,1\t1,2\t
+2,1\t2,2\t
+`);
+        });
+
+        it('should collect text from select', () => {
+            const testCase = elm.shadowRoot.querySelector('.select-testcase');
+
+            expect(testCase.innerText).toBe(`Chose a pet:
```

Comment:
in safari, `getComputedStyle(<option>) === 'inline'` while in other major browsers is block.